core/capabilities/targets/write_target.go

```diff
@@ -32,26 +34,40 @@ type WriteTarget struct {
 
 	lggr logger.Logger
 
+	// options for configuring local node info retrieval
+	registry         core.CapabilitiesRegistry
+	localNodeRetryMs int // how often to retry fetching local node info
+	localNode        *capabilities.Node
+
 	bound bool
 }
 
-func NewWriteTarget(lggr logger.Logger, id string, cr commontypes.ContractReader, cw commontypes.ChainWriter, forwarderAddress string) *WriteTarget {
+func NewWriteTarget(lggr logger.Logger, id string, cr commontypes.ContractReader, cw commontypes.ChainWriter, forwarderAddress string, registry core.CapabilitiesRegistry, localNodeInfoIntervalMs int) *WriteTarget {
 	info := capabilities.MustNewCapabilityInfo(
 		id,
 		capabilities.CapabilityTypeTarget,
 		"Write target.",
 	)
 
 	logger := lggr.Named("WriteTarget")
+	if localNodeInfoIntervalMs == 0 {
+		localNodeInfoIntervalMs = 5000
+	}
 
-	return &WriteTarget{
+	wt := &WriteTarget{
 		cr,
 		cw,
 		forwarderAddress,
 		info,
 		logger,
+		registry,
+		localNodeInfoIntervalMs,
+		nil,
 		false,
 	}
+
+	wt.ResolveLocalNodeInfo(context.Background())
```

Comment:
using background context risks leaving the thread started in ResolveLocalNodeInfo running preventing process shutdown whilst retrying